api/scpca_portal/models/project.py

```diff
@@ -409,37 +386,42 @@ def create_computed_file(future):
                 tasks.submit(
                     ComputedFile.get_project_merged_file,
                     self,
-                    single_cell_file_mapping,
-                    single_cell_workflow_versions,
+                    file_mappings_by_format[file_format].get(Sample.Modalities.SINGLE_CELL),
+                    workflow_versions_by_modality.get(Sample.Modalities.SINGLE_CELL),
                     file_format,
-                ).add_done_callback(create_computed_file)
+                ).add_done_callback(create_project_computed_file)
+
+                if file_mappings_by_format[file_format].get(Sample.Modalities.MULTIPLEXED):
+                    # We want a single ZIP archive for a multiplexed samples project.
+                    file_mappings_by_format[file_format].get(Sample.Modalities.MULTIPLEXED).update(
+                        file_mappings_by_format[file_format].get(Sample.Modalities.SINGLE_CELL)
+                    )
 
-                if multiplexed_file_mapping.get(file_format):
                     tasks.submit(
                         ComputedFile.get_project_multiplexed_file,
                         self,
-                        multiplexed_file_mapping,
-                        multiplexed_workflow_versions,
+                        file_mappings_by_format[file_format].get(Sample.Modalities.MULTIPLEXED),
+                        workflow_versions_by_modality[Sample.Modalities.MULTIPLEXED],
                         file_format,
-                    ).add_done_callback(create_computed_file)
+                    ).add_done_callback(create_project_computed_file)
 
-                if single_cell_file_mapping.get(file_format):
+                if file_mappings_by_format[file_format].get(Sample.Modalities.SINGLE_CELL):
```

Comment:
```suggestion
                if single_cell_file_mapping := file_mappings_by_format[file_format].get(Sample.Modalities.SINGLE_CELL):
```

I think this is a good place to use the :walrus: operator. 😎